### PR TITLE
Clarify outlier/low variability detection warnings ES

### LINKED ISSUE #6262: Confusing error message for deactivated observations
The error message given when observations are "missing" should be more informative/understandable for users, ref https://equinor.slack.com/archives/CDPJULWR4/p1696423905046869 . 

In `_es_update.py`:

```python
def _load_observations_and_responses(
    source_fs: EnsembleReader,
    obs: EnkfObs,
    alpha: float,
    std_cutoff: float,
    global_std_scaling: float,
    ens_mask: npt.NDArray[np.bool_],
    selected_observations: List[Tuple[str, Optional[List[int]]]],
) -> Any:
    ens_active_list = tuple(np.flatnonzero(ens_mask))

    S, observations, errors, obs_keys = _get_obs_and_measure_data(
        obs,
        source_fs,
        selected_observations,
        ens_active_list,
    )

    # Inflating measurement errors by a factor sqrt(global_std_scaling) as shown
    # in for example evensen2018 - Analysis of iterative ensemble smoothers for
    # solving inverse problems.
    # `global_std_scaling` is 1.0 for ES.
    errors *= sqrt(global_std_scaling)

    ens_mean = S.mean(axis=1)
    ens_std = S.std(ddof=0, axis=1)

    ens_std_mask = ens_std > std_cutoff
    ens_mean_mask = abs(observations - ens_mean) <= alpha * (ens_std + errors)

    obs_mask = np.logical_and(ens_mean_mask, ens_std_mask)

    update_snapshot = UpdateSnapshot(
        obs_name=obs_keys,
        obs_value=observations,
        obs_std=errors,
        obs_mask=obs_mask,
        response_mean=ens_mean,
        response_std=ens_std,
    )
    for missing_obs in obs_keys[~obs_mask]:
        _logger.warning(f"Deactivating observation: {missing_obs}")

    return S[obs_mask], (
        observations[obs_mask],
        errors[obs_mask],
        update_snapshot,
    )
```

### PR DESCRIPTION
**Pull request by yngve-sk**
**Issue**
Resolves #6262

### CODE REVIEW

**Comment by oyvindeide** [comment]
Note that some of this information is also available in the update snapshot: https://github.com/equinor/ert/blob/9f7ac75b4a8a6285659bb1ba641c14a2f9b79ec7/src/ert/analysis/_es_update.py#L60-L67 which is written out after an update, so might make sense to gather this information more.

**Comment by yngve-sk** [comment]
> Note that some of this information is also available in the update snapshot:
> 
> https://github.com/equinor/ert/blob/9f7ac75b4a8a6285659bb1ba641c14a2f9b79ec7/src/ert/analysis/_es_update.py#L60-L67
> 
> which is written out after an update, so might make sense to gather this information more.

I see, will try to gather it there, I think it also then makes sense to silence the original warning, ref this [slack convo](https://equinor.slack.com/archives/CDPJULWR4/p1696423905046869) where user gets many of these messages and wonders why they were disabled.

File: src/ert/analysis/_es_update.py [hunk 1]
```diff
@@ -401,8 +401,27 @@ def _load_observations_and_responses(
...
+    if np.any(~obs_mask):
+        fails_std_cut = np.where(~obs_mask & ~ens_std_mask & ens_mean_mask)[0]
+        fails_mean_cut = np.where(~obs_mask & ens_std_mask & ~ens_mean_mask)[0]
+
+        if len(fails_mean_cut) > 0:
+            _logger.warning(
+                f"Discarding outlier observations:"
+                f" [{[obs_keys[i] for i in fails_mean_cut ]}."
+                f" You may tune the alpha parameter to change this cutoff by"
+                f" setting  the ENKF_ALPHA parameter in the ert config."
```
Comment (andreas-el) [inline src/ert/analysis/_es_update.py:413]: Extra space after first word on this line

**Comment by oyvindeide** [comment]
Would be great to add this in the update snapshot instead of in the log, so we get a column there with the index.

File: src/ert/analysis/_es_update.py [hunk 2]
```diff
@@ -887,24 +910,39 @@ def _write_update_report(path: Path, snapshot: SmootherSnapshot, run_id: str) ->
...
+
+                if "name" in coord_cpy:
+                    del coord_cpy["name"]
+
+                if "time" in coord_cpy:
+                    assert isinstance(coord_cpy["time"], date)
+                    coord_cpy["time"] = coord_cpy["time"].strftime("%Y-%m-%d")
+
+                obs_coord_str = ", ".join(
+                    f"{k}:{v}" for k, v in coord_cpy.items() if k != "name"
```
Comment (oyvindeide) [inline src/ert/analysis/_es_update.py:942]: Checking for key here, even though it is deleted.

Comment (oyvindeide) [inline src/ert/analysis/_es_update.py:942]: Is it possible to do this when extracting it in the first place? And just return the completed `index` string instead? Think that makes it easier to read.

Comment (oyvindeide) [reply src/ert/analysis/_es_update.py:942]: That would also make it easier to include it in the update report in the gui:
image

**Review by oyvindeide — CHANGES_REQUESTED** [review]
Looks like a good change! Should also include it in the update report in the gui, and some other small comments.

**Comment by sondreso** [comment]
Closing this as the original issue has been fixed: https://github.com/equinor/ert/issues/6262

**Pull request closed without merging**